src/FsAutoComplete.Core/AdaptiveExtensions.fs

```diff
@@ -19,35 +19,40 @@ module AdaptiveExtensions =
 
     member cts.TryCancel() =
       try
-        if not <| isNull cts then
-          cts.Cancel()
+        if not <| isNull cts then cts.Cancel()
       with
       | :? ObjectDisposedException
       | :? NullReferenceException -> ()
 
     member cts.TryDispose() =
       try
-        if not <| isNull cts then
-          cts.Dispose()
-      with _ ->
-        ()
+        if not <| isNull cts then cts.Dispose()
+      with
+      | _ -> ()
 
 
   type TaskCompletionSource<'a> with
 
     /// https://github.com/dotnet/runtime/issues/47998
-    member tcs.TrySetFromTask(real: Task<'a>) =
+    member tcs.TrySetFromTaskFinished(real: Task<'a>) =
 
       // note: using ContinueWith instead of task CE for better stack traces
-      real.ContinueWith(fun (task: Task<_>) ->
+      real.ContinueWith (fun (task: Task<_>) ->
+#if NET9_0_OR_GREATER
+        tcs.TrySetFromTask(task) |> ignore<bool>
```

Comment:
oh fascinating - so the runtime is generally getting APIs that are better/correct/easier to use?